### ISSUE TITLE
make the lib multi-platform

### ISSUE THREAD

**Issue by oshai** [MicroUtils/kotlin-logging#21]

**Comment by LDVSOFT** [comment]
I would really like this awesome library supporting multiplatform code. There is already attempt using `mu.kotlin-logging` [here: https://github.com/shafirov/klogging](https://github.com/shafirov/klogging), but it does not provide API to be used in multiplatform code.

**Comment by oshai** [comment]
Do you have an idea or link how to do that? I am not sure what is the level of support in kotlin for such libs.

**Comment by LDVSOFT** [comment]
I referenced only ideas. There is [a discussion](https://discuss.kotlinlang.org/t/multi-platform-module-recipes/3058/6) on supporting the multi-platform projects.

**Comment by oshai** [comment]
Thanks for the links, seems like not that stable at the moment... So I am still waiting for JB to announce of something official.

**Comment by LDVSOFT** [comment]
Yeah, official support would be really nice. Thank you for all your effort!

**Comment by oshai** [comment]
When its released, a pull request will be nice ;-)

On 16 Aug 2017 12:06, "Lapshin Dmitry" <notifications@github.com> wrote:

Yeah, official support would be really nice. Thank you for all your effort!

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/MicroUtils/kotlin-logging/issues/21#issuecomment-322711055>,
or mute the thread

**Comment by oshai** [comment]
It will be possible in 1.2: http://kotlinlang.org/docs/reference/multiplatform.html

**Comment by Lewik** [comment]
https://github.com/Lewik/klogging can jvm, js and common logging.

**Comment by oshai** [comment]
changes for this issue are on the multiplatform branch: https://github.com/MicroUtils/kotlin-logging/tree/multiplatform

**Comment by oshai** [comment]
release 1.5.3 is from multiplatform branch, currently only jvm version but you are welcome to try.

**Comment by oshai** [comment]
All work was merged to master. There is no js artifact yet but repo is ready for that.

**Comment by joelhandwell** [comment]
@oshai Is this also applicable for [Kotlin/Native multi platform projects](https://github.com/JetBrains/kotlin-native/blob/master/MULTIPLATFORM.md)?

**Comment by oshai** [comment]
No, currently this issue is only related to jvm/js. But it is true that in the future we should have native flavour as well.

**Comment by oshai** [comment]
@LDVSOFT @Lewik - if you can test common + js from here: https://github.com/MicroUtils/kotlin-logging/releases/tag/1.5.5 it will be great. Also, any comments on the implementation are welcome.

**Comment by oshai** [comment]
I also publisher common and js to bintray/jcenter. feedback is welcome.

**Comment by anidotnet** [comment]
Currently maven/gradle build is failing for kotlin/jvm project as it is not able to find the common module.

```
 > Could not find io.github.microutils:kotlin-logging-common:1.5.5.
     Searched in the following locations:
         file:/home/anindya/.m2/repository/io/github/microutils/kotlin-logging-common/1.5.5/kotlin-logging-common-1.5.5.pom
         file:/home/anindya/.m2/repository/io/github/microutils/kotlin-logging-common/1.5.5/kotlin-logging-common-1.5.5.jar
         https://repo.maven.apache.org/maven2/io/github/microutils/kotlin-logging-common/1.5.5/kotlin-logging-common-1.5.5.pom
         https://repo.maven.apache.org/maven2/io/github/microutils/kotlin-logging-common/1.5.5/kotlin-logging-common-1.5.5.jar
         https://oss.sonatype.org/content/repositories/snapshots/io/github/microutils/kotlin-logging-common/1.5.5/kotlin-logging-common-1.5.5.pom
         https://oss.sonatype.org/content/repositories/snapshots/io/github/microutils/kotlin-logging-common/1.5.5/kotlin-logging-common-1.5.5.jar

```
Do you have any plan to publish the common artifact to central maven repo?

**Comment by oshai** [comment]
I was hoping I will not have to, it will break usage for existing users trying to upgrade.
I will try to define source-sets to overcome this issue.

**Comment by oshai** [comment]
@anidotnet - thanks for reporting. seems something was changed between Kotlin versions 1.2.30->1.2.40 in pom generation. I will try to ask around.
In the meanwhile, I created version 1.5.6 that should be ok. It will be great if you can verify that.

**Comment by oshai** [comment]
opened an issue for it: https://youtrack.jetbrains.com/issue/KT-25799

**Comment by anidotnet** [comment]
@oshai I still could not see the new version 1.5.6 in central maven repo.

**Comment by oshai** [comment]
Please vote for this issue, as it prevents us from upgrading kotlin version: https://youtrack.jetbrains.com/issue/KT-25799

**Comment by oshai** [comment]
I published version 1.5.9 to maven central which I hope will solve the issue with missing artifacts. There is still a broken build in Travis that I am handling because of js tests issue. If you see any other errors please open a new issue.
